src/Microsoft.Extensions.FileProviders.Abstractions/NullFileProvider/NullDirectoryContents.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.FileProviders.Abstractions
+{
+    public class NullDirectoryContents : IDirectoryContents
+    {
+        public bool Exists => false;
+
+        public IEnumerator<IFileInfo> GetEnumerator()
+        {
+            yield break;
```

Comment:
That's what I mentioned above. But we need to change the layering. We need to move NullFileProvider to its own project or else we'll get a circular dependency.